Ruby/Ruby.sublime-syntax

```diff
@@ -479,11 +533,30 @@ contexts:
       scope: variable.other.constant.ruby
 
   well-known-methods:
-    - match: '\b(initialize|new|loop|include|extend|prepend|raise|fail|attr_reader|attr_writer|attr_accessor|attr|catch|throw|module_function|public|protected|private)\b(?![?!])'
+    # exceptions
+    - match: \bcatch\b(?![?!])
+      scope: keyword.control.exception.catch.ruby
+      push: function-call-arguments
+    # flow
+    - match: \braise\b(?![?!])
+      scope: keyword.control.flow.raise.ruby
+      push: function-call-arguments
+    - match: \bfail\b(?![?!])
+      scope: keyword.control.flow.fail.ruby
```

Comment:
Does this behave like `die` or `panic` in other languages?

Reply:
I don't know. I just followed the rule to add such kind of flow control keywords to the group of `keyword.control.flow`

Reply:
Sorry for jumping in (just stumbled in here), but if the question of @FichteFoll was related to the `fail` keyword:
It is a synonym for `raise`, which is like "throw Exception" in other languages.

Reply:
The `die` or `panic` functions in Perl are nothing else. They just cause the process to die just like `exit` in some other syntaxes. The point just is ... Which keywords to treat as synonymes and which not. 

The same question arises with the `redo`, `next`, `retry` keywords above. All are some kind of `continue` synonym but with little different meanding?

Reply:
Ah okay. Well I feel like `raise` should be the original and `fail` the synonym as the latter is much less common.
Whereas `next` and `retry` for example are not synonyms for me (except from the point of syntax color... which probably is the point here).